File: src/main/java/com/truveta/opentoken/tokentransformer/TokenTransformer.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.tokentransformer;

/**
 * A generic interface for the token transformer.
 */
public interface TokenTransformer {
    /**
     * Transforms the token using a token transformation rule/strategy.
     * 
     * @param token the token to be transformed.
     * 
     * @return the transformed token.
     * 
     * @throws Exception error encountered while transforming the token.
     */
    String transform(String token) throws Exception;
}

File: lib/nodejs/opentoken/src/tokentransformer/TokenTransformer.ts
/**
 * Copyright (c) Truveta. All rights reserved.
 */

/**
 * Interface for token transformers.
 */
export interface TokenTransformer {
  /**
   * Transforms a token.
   *
   * @param token - The token to transform.
   * @returns The transformed token.
   */
  transform(token: string): string;
}
